hcl/hcl.go

```diff
@@ -26,73 +26,73 @@ type Blocks interface {
 
 // NOTE(dcohen) this is currently a separate config block, as opposed to a parent block of the others
 type Agent struct {
-	Redactions []Redact `hcl:"redact,block"`
+	Redactions []Redact `hcl:"redact,block" json:"redactions,omitempty"`
 }
 
 type Host struct {
-	Commands     []Command     `hcl:"command,block"`
-	Shells       []Shell       `hcl:"shell,block"`
-	GETs         []GET         `hcl:"GET,block"`
-	Copies       []Copy        `hcl:"copy,block"`
-	DockerLogs   []DockerLog   `hcl:"docker-log,block"`
-	JournaldLogs []JournaldLog `hcl:"journald-log,block"`
-	Excludes     []string      `hcl:"excludes,optional"`
-	Selects      []string      `hcl:"selects,optional"`
-	Redactions   []Redact      `hcl:"redact,block"`
+	Commands     []Command     `hcl:"command,block" json:"commands,omitempty"`
+	Shells       []Shell       `hcl:"shell,block" json:"shells,omitempty"`
+	GETs         []GET         `hcl:"GET,block" json:"gets,omitempty"`
+	Copies       []Copy        `hcl:"copy,block" json:"copies,omitempty"`
+	DockerLogs   []DockerLog   `hcl:"docker-log,block" json:"docker_log,omitempty"`
+	JournaldLogs []JournaldLog `hcl:"journald-log,block" json:"journald_log,omitempty"`
+	Excludes     []string      `hcl:"excludes,optional" json:"excludes,omitempty"`
+	Selects      []string      `hcl:"selects,optional" json:"selects,omitempty"`
+	Redactions   []Redact      `hcl:"redact,block" json:"redactions,omitempty"`
 }
 
 type Product struct {
-	Name         string        `hcl:"name,label"`
-	Commands     []Command     `hcl:"command,block"`
-	Shells       []Shell       `hcl:"shell,block"`
-	GETs         []GET         `hcl:"GET,block"`
-	Copies       []Copy        `hcl:"copy,block"`
-	DockerLogs   []DockerLog   `hcl:"docker-log,block"`
-	JournaldLogs []JournaldLog `hcl:"journald-log,block"`
-	Excludes     []string      `hcl:"excludes,optional"`
-	Selects      []string      `hcl:"selects,optional"`
-	Redactions   []Redact      `hcl:"redact,block"`
+	Name         string        `hcl:"name,label" json:"name"`
+	Commands     []Command     `hcl:"command,block" json:"commands,omitempty"`
+	Shells       []Shell       `hcl:"shell,block" json:"shells,omitempty"`
+	GETs         []GET         `hcl:"GET,block" json:"gets,omitempty"`
+	Copies       []Copy        `hcl:"copy,block" json:"copies,omitempty"`
+	DockerLogs   []DockerLog   `hcl:"docker-log,block" json:"docker_log,omitempty"`
+	JournaldLogs []JournaldLog `hcl:"journald-log,block" json:"journald_log,omitempty"`
+	Excludes     []string      `hcl:"excludes,optional" json:"excludes,omitempty"`
+	Selects      []string      `hcl:"selects,optional" json:"selects,omitempty"`
+	Redactions   []Redact      `hcl:"redact,block" json:"redactions,omitempty"`
 }
 
 type Redact struct {
-	Label   string `hcl:"name,label"`
-	ID      string `hcl:"id,optional"`
+	Label   string `hcl:"name,label" json:"label"`
+	ID      string `hcl:"id,optional" json:"ID"`
```

Comment:
Was this ID supposed to be lowercase?

Reply:
I went with a capitalized acronym here, and JSON style guides conveniently seem to not mention acronym capitalization. I talked this over with a resident json/js expert (my partner) and he also said lowercase it. It makes it consistent with all the other fields, so let's go with that.